src/components/Repository/FileList.js

```diff
@@ -272,7 +274,12 @@ class FileList extends Component {
                 defaultHidden: false, 
                 getCellValue: row => { 
                     return row['redcap_id'] !== "Multiple Participants" 
-                    ? <button onClick={(e) => this.props.props.history.push('/report')} type='button' data-toggle="tooltip" data-placement="top" title="View participant information" className='table-column btn btn-link p-0'>{row["redcap_id"]}</button>
+                    ? <button onClick={async (e) => {
+                            await setSummaryDatasets(row['redcap_id'][0]).then(() => {
+                                console.log(store.getState());
+                                this.props.props.history.push('/report');

```

Comment:
Hmm...so what does this look like? We don't see the report page until the data is loaded? This seems not user friendly. Ideally we would go to the page, and then load the data and have spinners if it takes a while. 
I don't think we will have an issue with performance given the data we are getting, but this could look like the app is hanging to a user and invite them to click multiple times on the link, causing many requests to the server.

Reply:
Yeah it basically takes no time at all for the data to load, it just takes even _less_ time to get to the /report page, so it ends up showing "No Data." A spinner is a good idea, I'll try that!